src/S3/PostObjectV4.php

```diff
@@ -193,7 +175,7 @@ private function generateUri()
 
     protected function getPolicyAndSignature(CredentialsInterface $creds)
```

Comment:
This method needs to modify the policy to add the appropriate credential, algorithm, and date to the list of conditions.